src/diffusers/configuration_utils.py

```diff
@@ -157,7 +157,12 @@ def save_config(self, save_directory: Union[str, os.PathLike], push_to_hub: bool
         os.makedirs(save_directory, exist_ok=True)
 
         # If we save using the predefined names, we can load using `from_config`
-        output_config_file = os.path.join(save_directory, self.config_name)
+        filename = kwargs.pop("filename", None)
```

Comment:
Uff not too happy about such fundamental changes. Why is this needed?

Reply:
Lets us pass a filename to save a workflow file. Users can always manually rename files.